src/libfuncs/array.rs

```diff
@@ -1154,16 +959,13 @@ fn assert_nonnull<'ctx, 'this>(
         ods::llvm::ptrtoint(context, IntegerType::new(context, 64).into(), ptr, location).into(),
```

Comment:
I'm sure I understand, you mean doing ?

```rust
fn assert_nonnull<'ctx, 'this>(
    context: &'ctx Context,
    entry: &'this Block<'ctx>,
    location: Location<'ctx>,
    ptr: Value<'ctx, 'this>,
    msg: &str,
) -> Result<()> {
    let k0 = entry.const_int(context, location, 0, 64)?;

    let ptr_is_not_null = entry.append_op_result(
        ods::llvm::icmp(
            context,
            IntegerType::new(context, 64).into(),
            ptr,
            k0,
            IntegerAttribute::new(IntegerType::new(context, 1).into(), 1).into(),
            location,
        )
        .into(),
    )?;

    entry.append_operation(cf::assert(context, ptr_is_not_null, msg, location));
    Ok(())
}
```

Reply:
Almost, I think it would be more like this:
```rs
ods::llvm::icmp(
    context,
    llvm::r#type::pointer(context, 0),                                     // We're operating with pointers so this should be a pointer.
    ptr,                                                                   // Our value.
    null_ptr,                                                              // The value we're comparing with.
    IntegerAttribute::new(IntegerType::new(context, 64).into(), 0).into(), // I think the `eq` value is zero.
    location
).into()
```

Reply:
Just proposed something, should resolve this, let me know